lib/ops/resize/kernel.glsl

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license MIT
+ * @author Arkadiy Pilguk(apilguk@gmail.com)
+ * @author Mihail Zachepilo(mihailzachepilo@gmail.com)
+ * Copyright 2018 Peculiar Ventures and Pentatonica.
+ * All rights reserved.
+ */
+
+vec4 operation(float y, float x) {
+  vec4 value = vec4(0.0);
+
+  for (float dx = 0.0; dx < ceil(H); dx += 1.0) {
+    for (float dy = 0.0; dy < ceil(W); dy += 1.0) {
+      vec4 v = pickValue_tSrc((y * H) + dy, (x * W) + dx);
+
+      if (S == 0.0) {
+        value = v;
+      }
+
+      if (S == 1.0) {
+        value += v;
+      }
+    }
+  }
+
+  if (S == 1.0) {
+    value /= W * H;
+  }
```

Comment:
Changin this to:
```suggestion
  if (S == 1.0) {
    value /= ceil(W) * ceil(H);
  }
```
It will fix points 2 and 3 from the main comment.
If apply this solution we could drop the mention that upsizes works incorrectly and just mention that the top-left corner will be the "center of mass". Alternatively, we could add ann API to change the transform-origin, but this will slightly extend this operation.